uefi/src/proto/shell/mod.rs

```diff
@@ -2,12 +2,208 @@
+    /// * `None` - If environment variable does not exist
+    #[must_use]
+    pub fn get_env(&self, name: &CStr16) -> Option<&CStr16> {
+        let name_ptr: *const Char16 = core::ptr::from_ref::<CStr16>(name).cast();
+        let var_val = unsafe { (self.0.get_env)(name_ptr.cast()) };
+        if var_val.is_null() {
+            None
+        } else {
+            unsafe { Some(CStr16::from_ptr(var_val.cast())) }
+        }
+    }
+
+    /// Gets the list of environment variables
+    ///
+    /// # Returns
+    ///
+    /// * `Vec<env_names>` - Vector of environment variable names
+    #[must_use]
+    pub fn get_envs(&self) -> Vars {
+        let env_ptr = unsafe { (self.0.get_env)(ptr::null()) };
+        Vars {
+            inner: env_ptr.cast::<Char16>(),
+            placeholder: PhantomData,
+        }
+    }
+
+    /// Sets the environment variable
+    ///
+    /// # Arguments
+    ///
+    /// * `name` - The environment variable for which to set the value
+    /// * `value` - The new value of the environment variable
+    /// * `volatile` - Indicates whether the variable is volatile or
+    ///   not
+    ///
+    /// # Returns
+    ///
+    /// * `Status::SUCCESS` - The variable was successfully set
+    pub fn set_env(&self, name: &CStr16, value: &CStr16, volatile: bool) -> Status {
+        let name_ptr: *const Char16 = core::ptr::from_ref::<CStr16>(name).cast();
+        let value_ptr: *const Char16 = core::ptr::from_ref::<CStr16>(value).cast();
+        unsafe { (self.0.set_env)(name_ptr.cast(), value_ptr.cast(), volatile) }
+    }
+
+    /// Returns the current directory on the specified device
+    ///
+    /// # Arguments
+    ///
+    /// * `file_system_mapping` - The file system mapping for which to get
+    ///   the current directory
+    ///
+    /// # Returns
+    ///
+    /// * `Some(cwd)` - CStr16 containing the current working directory
+    /// * `None` - Could not retrieve current directory
+    #[must_use]
+    pub fn get_cur_dir(&self, file_system_mapping: Option<&CStr16>) -> Option<&CStr16> {
+        let mapping_ptr: *const Char16 = file_system_mapping.map_or(ptr::null(), |x| (x.as_ptr()));
+        let cur_dir = unsafe { (self.0.get_cur_dir)(mapping_ptr.cast()) };
+        if cur_dir.is_null() {
+            None
+        } else {
+            unsafe { Some(CStr16::from_ptr(cur_dir.cast())) }
+        }
+    }
+
+    /// Changes the current directory on the specified device
+    ///
+    /// # Arguments
+    ///
+    /// * `file_system` - Pointer to the file system's mapped name.
+    /// * `directory` - Points to the directory on the device specified by
+    ///   `file_system`.
+    ///
+    /// # Returns
+    ///
+    /// * `Status::SUCCESS` - The directory was successfully set
+    ///
+    /// # Errors
+    ///
+    /// * `Status::EFI_NOT_FOUND` - The directory does not exist
+    pub fn set_cur_dir(&self, file_system: Option<&CStr16>, directory: Option<&CStr16>) -> Status {
```

Comment:
`set_current_dir` to match [std](https://doc.rust-lang.org/std/env/fn.set_current_dir.html)